### PR TITLE
lint our markdown

### PR DESCRIPTION
**Pull request by ljharb**
From https://github.com/airbnb/enzyme/pull/984#issuecomment-309115597

Now our docs will be linted.

### CODE REVIEW

File: .eslintrc-markdown [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "root": true,
+  "plugins": [
+    "markdown"
+  ],
+  "rules": {
+    "class-methods-use-this": 0,
+    "import/no-unresolved": 0,
+    "import/no-extraneous-dependencies": 0,
+    "import/extensions": 0,
+    "max-len": 0,
```
Comment (nfcampos) [inline .eslintrc-markdown]: why disable this one?

**Comment by nfcampos** [comment]
only question is for me is what if someone adds a markdown with an extension other .md? should we do anything for that?

File: .eslintrc-markdown (hunk 1 above)
Comment (ljharb) [inline .eslintrc-markdown]: I don't really like this rule in general; it's basically a poor proxy for limiting complexity. All the lines it warned on weren't worth arbitrarily splitting up.

**Comment by ljharb** [comment]
I'm not sure if there's anyway we can prevent that short of manual review.

**Pull request merged**